libs/stats/odc/stats/_gm_ls_bitmask.py

```diff
@@ -0,0 +1,172 @@
+            |   |   o---------------------------------------------------------> snow_ice_confidence
+            |   x-------------------------------------------------------------> cirrus_confidence
+            0-----------------------------------------------------------------> cirrus_confidence
+        """
+        mask_band = xx[self.mask_band]
+        xx = xx.drop_vars([self.mask_band])
+
+        # set cloud_mask (dilated_cloud + cloud + cloud_shadow) bitmask - True=cloud, False=non-cloud
+        cloud_mask = da.bitwise_and(mask_band, 0b0000_0000_0001_1010) != 0
+
+        # set no_data bitmask - True=data, False=no-data
+        keeps = da.bitwise_and(mask_band, 0b0000_0000_0000_0001) == 0
+
+        # drops nodata pixels and add cloud_mask from xx
+        xx = keep_good_only(xx, keeps)
+        xx["cloud_mask"] = cloud_mask
+
+        return xx
+
+    def input_data(self, task: Task) -> xr.Dataset:
+
+        chunks = {"y": -1, "x": -1}
+
+        xx = load_with_native_transform(
+            task.datasets,
+            bands=self.bands + [self.mask_band],
+            geobox=task.geobox,
+            native_transform=self._native_tr,
+            fuser=self._fuser,
+            groupby="solar_day",
+            resampling=self.resampling,
+            chunks=chunks,
+        )
+
+        return xx
+
+    def reduce(self, xx: xr.Dataset) -> xr.Dataset:
+        scale = 0.0000275
+        offset = -0.2
+        return_SR = False
+        cfg = dict(
+            maxiters=1000,
+            num_threads=1,
+            scale=scale,
+            offset=offset,
+            return_SR=return_SR,
+            reshape_strategy="mem",
+            out_chunks=(-1, -1, -1),
+            work_chunks=self.work_chunks,
+            compute_count=True,
+            compute_mads=True,
+        )
+
+        cloud_mask = xx["cloud_mask"]
+        xx = xx.drop_vars(["cloud_mask"])
+
+        # erase pixels with cloud
+        xx = erase_bad(xx, cloud_mask)
+
+        gm = geomedian_with_mads(xx, **cfg)
+        gm = gm.rename(self.renames)
+
+        # Scale USGS Landsat bands into surface reflectance
+        if return_SR:
+            for band in gm.data_vars:
+                if band in self.bands:
+                    gm[band] = (scale*10000 * gm[band]+offset*10000).astype(gm[band].dtype)
+        
+        # TODO: handle edev scaling correctly
+        # need to investigate what stats is doing 
+        
+        return gm
+
+    def _fuser(self, xx):
+        """
+        Fuse cloud_mask with OR, and apply mask_cleanup if requested
+        """
+        cloud_mask = xx["cloud_mask"]
+        xx = _xr_fuse(xx.drop_vars(["cloud_mask"]), partial(_first_valid_np, nodata=0), '')
+        xx["cloud_mask"] = _xr_fuse(cloud_mask, _fuse_or_np, cloud_mask.name)
+
+        # apply filters - [r1, r2]
+        # r1 = shrinks away small areas of the mask
+        # r2 = adds padding to the mask
+        if self.filters is not None:
```

Comment:
@NikitaGandhi @alexgleith the fusing may not work as expected here. It turns out that `fuser` is called twice during `load_with_native_transform` so the filtering is also being applied twice which is probably not the intended behaviour. The same (potential?) issue also happens in `load_enum_filtered`.

We can avoid this by moving the filtering into `reduce`